.github/workflows/validate.yml

```diff
@@ -0,0 +1,31 @@
+name: validate
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - lint
```

Comment:
The official `golangci-lint`  GitHub Action can also run on Windows, Darwin. Wondering if we can trick it by setting GOOS/GOARCH to also lint non-matching platforms (especially for the CLI, which will be extensively used on macOS and Windows)

Reply:
Yes I think we can have a special matrix to run Windows and MacOS runners in this case.

Reply:
Yeah, it's tricky, because of course we can't run a macOS container, and Windows doesn't support BuildKit for Windows containers (if it would, it's still possible that building takes a _long time_ (Moby Ci can take 10 minutes to build the Dockerfile, a large portion of that is taken by just pulling the base image 😞).

So if it's possible to somehow trick golangci-lint (through `GOOS` / `GOARCH`) that would be a lot easier.

If not, then perhaps (for CI) we would have to consider using the official golangci-lint GitHub action, which doesn't (I think?) run containerised, but at least runs fast (and on Windows and macOS). We can still use the BuildKit approach for local use (and for Linux perhaps).

Reply:
@thaJeztah 

> using the official golangci-lint GitHub action

Yes I think we can do that with https://github.com/golangci/golangci-lint-action for Win/MacOS.